integrations/lib/multiwoven/integrations/destination/mailchimp/client.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+module Multiwoven
+  module Integrations
+    module Destination
+      module Mailchimp
+        include Multiwoven::Integrations::Core
+
+        API_VERSION = "3.0"
+
+        class Client < DestinationConnector
+          prepend Multiwoven::Integrations::Core::RateLimiter
+
+          def check_connection(connection_config)
+            connection_config = connection_config.with_indifferent_access
+            initialize_client(connection_config)
+            authenticate_client
+            success_status
+          rescue StandardError => e
+            failure_status(e)
+          end
+
+          def discover(_connection_config = nil)
+            catalog = build_catalog(load_catalog)
+            catalog.to_multiwoven_message
+          rescue StandardError => e
+            handle_exception(e, {
+                               context: "MAILCHIMP:DISCOVER:EXCEPTION",
+                               type: "error"
+                             })
+          end
+
+          def write(sync_config, records, action = "create")
+            @sync_config = sync_config
+            @action = sync_config.stream.action || action
```

Comment:
@xyfer17 Minor nitpick, is `@action` used for anything?

Reply:
fixed in latest commit